registry/repositoryClient.go

```diff
@@ -0,0 +1,163 @@
+package registry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	artifactspec "github.com/oras-project/artifacts-spec/specs-go/v1"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras-go/v2/registry/remote"
+)
+
+const (
+	maxBlobSizeLimit     = 32 * 1024 * 1024 // 32 MiB
+	maxManifestSizeLimit = 4 * 1024 * 1024  // 4 MiB
+)
+
+// repositoryClient implements Repository
+type repositoryClient struct {
+	remote.Repository
+}
+
+// Resolve resolves a reference(tag or digest) to a manifest descriptor
+func (c *repositoryClient) Resolve(ctx context.Context, reference string) (ocispec.Descriptor, error) {
+	return c.Repository.Resolve(ctx, reference)
+}
+
+// ListSignatures returns signature manifests filtered by fn given the
+// artifact manifest descriptor
+func (c *repositoryClient) ListSignatures(ctx context.Context, desc ocispec.Descriptor, fn func(signatureManifests []ocispec.Descriptor) error) error {
+	if err := c.Repository.Referrers(ctx, ocispec.Descriptor{
+		Digest: desc.Digest,
+	}, ArtifactTypeNotation, func(referrers []artifactspec.Descriptor) error {
+		var sigManifestDesc []ocispec.Descriptor
+		for _, referrer := range referrers {
+			sigManifestDesc = append(sigManifestDesc, ocispecDescriptorFromArtifact(referrer))
+		}
+		return fn(sigManifestDesc)
+	}); err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return c.Repository.Referrers(ctx, ocispec.Descriptor{
		Digest: desc.Digest,
	}, ArtifactTypeNotation, func(referrers []artifactspec.Descriptor) error {
		var sigManifestDesc []ocispec.Descriptor
		for _, referrer := range referrers {
			sigManifestDesc = append(sigManifestDesc, ocispecDescriptorFromArtifact(referrer))
		}
		return fn(sigManifestDesc)
	})
```

Basically, code structure
```go
if err := do(); err != nil {
    return err
}
return nil
```
can be simplified as
```go
return do()
```